src/casp/services/api/server.py

```diff
@@ -35,23 +37,28 @@ async def __get_embeddings(myfile) -> t.Tuple[t.List, "numpy.array"]:
     return query_ids, embeddings
 
 
-def __get_appx_knn_matches(embeddings):
-    index_endpoint = aiplatform.MatchingEngineIndexEndpoint(index_endpoint_name=settings.KNN_SEARCH_ENDPOINT_ID)
+def __get_appx_knn_matches(embeddings, model_system_name: str):
+    cas_model = ops.get_model_by(system_name=model_system_name)
+    cas_matching_engine_index = cas_model.cas_matching_engine
+
+    index_endpoint = matching_engine_client.IncreasedGRPCSizeMatchingEngine(
+        index_endpoint_name=cas_matching_engine_index.endpoint_id
+    )
 
     response = index_endpoint.match(
-        deployed_index_id=settings.KNN_SEARCH_DEPLOYED_INDEX_ID,
+        deployed_index_id=cas_matching_engine_index.deployed_index_id,
         queries=embeddings,
         num_neighbors=settings.KNN_SEARCH_NUM_MATCHES,
     )
     return response
 
 
-def __get_cell_type_distribution(query_ids, knn_response):
+def __get_cell_type_distribution(query_ids, knn_response, model_system_name: str):
```

Comment:
Is this a good use of BQ? Why not calculate summary statistics locally? We already have the distances localized to memory. All we need is the metadata (here, cell type) of the hits.

Reply:
I agree with that! [Issue](https://github.com/cellarium-ai/cellarium-cloud/issues/96)